typing/typemod.ml

```diff
@@ -2097,22 +2097,31 @@ let wrap_constraint_with_shape env mark arg mty
 
 
 (* Summary for F(X) *)
-type application_summary = {
-  loc: Location.t;
-  attributes: attributes;
-  f_loc: Location.t; (* loc for F *)
-  arg_is_syntactic_unit: bool;
-  arg: Typedtree.module_expr;
-  arg_path: Path.t option;
-  shape: Shape.t
-}
-
-let simplify_app_summary app_view =
-  let mty = app_view.arg.mod_type in
-  match app_view.arg_is_syntactic_unit , app_view.arg_path with
-  | true,   _ -> Includemod.Error.Unit, mty
-  | false, Some p -> Includemod.Error.Named p, mty
-  | false, None -> Includemod.Error.Anonymous, mty
+type application_summary =
+  | Normal_app of {
+      loc: Location.t;
+      attributes: attributes;
+      f_loc: Location.t; (* loc for F *)
+      arg_is_syntactic_unit: bool;
+      arg: Typedtree.module_expr;
+      arg_path: Path.t option;
+      shape: Shape.t;
+    }
+  | Unit_app of {
+      loc: Location.t;
+      attributes: attributes;
+      f_loc: Location.t; (* loc for F *)
+    }
```

Comment:
I would have expected a more factorized type to avoid the `let Normal_app ... | Unit_app ...` gymnastics below.

```ocaml
type application_summary = {
  loc: Location.t;
  attributes: attributes;
  f_loc: Location.t; (* loc for F *)
  arg: argument_summary option (* None for () *);
}
and argument_summary = {
  is_syntactic_unit: bool;
  arg: Typedtree.module_expr;
  path: Path.t option;
  shape: Shape.t;
}
```

Are you convinced that your approach is nicer? (If yes, please keep it.)